external-crates/move/move-compiler/src/expansion/ast.rs

```diff
@@ -404,6 +404,16 @@ pub type Type = Spanned<Type_>;
 // Expressions
 //**************************************************************************************************
 
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum Mutability {
+    // mut x
+    Mut(Loc),
+    // x
+    Imm,
+    // Used for assignments or for legacy code without explicit mutability
+    NotApplicable,
```

Comment:
This variant seems a little odd to me. I appreciate what it's doing, but I'd prefer one of the other two options:

(a) feature-gating the actual checking code in typing using the Context feature versioning stuff (the context could even check and cache the flag for mutability checking) and then marking all `legacy` code as `Immutable` but never checking;

(b) or _always_ marking definitions made under pre-2024 stuff as `mutable` (since they kinda are), with no nedd to feature gate or handle this case in this branch.

Reply:
These are good suggestions. Let me attempt (b)